### PR TITLE
Fix PTR Record generation

### PR DESCRIPTION
**Pull request by sadlil**
Ref : https://github.com/kubernetes/dns/pull/25#issuecomment-304734753

### CODE REVIEW

File: pkg/dns/dns.go [hunk 1]
```diff
@@ -523,6 +523,23 @@ func getHostname(address *v1.EndpointAddress) (string, bool) {
 	return "", false
 }
 
+func (kd *KubeDNS) ensurePodSubdomain(address *v1.EndpointAddress, svc *v1.Service) bool {
```
Comment (thockin) [inline pkg/dns/dns.go]: How frequently does this run?  I mean, it runs on every pod that needs it, but what is the period for refresh?

I'm worried about swamping the apiserver when there are lots of DNS replicas active, but we really can't watch pods.

@bowei @smarterclayton

File: pkg/dns/dns_test.go [hunk 2]
```diff
@@ -427,6 +429,7 @@ func TestNamedHeadlessServiceEndpointUpdate(t *testing.T) {
 
 func TestNamedHeadlessServiceEndpointDelete(t *testing.T) {
```
Comment (thockin) [inline pkg/dns/dns_test.go]: Add a testcase for a en endpoint that requests reverse DNS but the pod doesn't confirm?

File: pkg/dns/dns.go (hunk 1 above)
Comment (feiskyer) [reply pkg/dns/dns.go]: Does adding the subdomain to endpoints make sense for this?  (not sure whether it is general enough for endpoints having a subdomain.)

**Comment by thockin** [comment]
@bowei shoot - this fell through again.

**Comment by chrislovecnm** [comment]
@thockin / @sadlil what is the status on this?  How do we not wait for 1.9 for this???

**Comment by bowei** [comment]
@chrislovecnm Are you saying to put in 1.8 or fix in 1.9?

**Comment by chrislovecnm** [comment]
@bowei we need a fixed container ASAP, but when can we get this released?

**Pull request closed without merging**